lib/rules/length-zero-no-unit/index.mjs

```diff
@@ -66,32 +68,33 @@ const rule = (primary, secondaryOptions, context) => {
 
 			if (!isWord(valueNode)) return;
 
-			const numberUnit = valueParser.unit(value);
+			const dimension = valueParser.unit(value);
+			const decompositionFailed = dimension === false;
+
+			if (decompositionFailed) return;
 
-			if (numberUnit === false) return;
+			const { number, unit } = dimension;
 
-			const { number, unit } = numberUnit;
+			if (!isZero(number)) return;
 
-			if (unit === '') return;
+			if (isNumber(unit)) return;
 
 			if (!isLength(unit)) return;
 
 			if (isFraction(unit)) return;
 
-			if (!isZero(number)) return;
-
-			if (context.fix) {
-				let regularNumber = number;
+			const fix = () => {
+				valueNode.value = number[0] === '.' ? number.slice(1) : number;
```

Comment:
[suggest] I prefer `startsWith` over `[0]` because of readability, like natural English, as the original code does:

```suggestion
				valueNode.value = number.startsWith('.') ? number.slice(1) : number;
```

Reply:
Just curious, how long does the `npm run build` take on your machine?

Reply:
My machine is MacBook with Apple M1. Here's the `time` command result. I feel about less than 10 seconds.

```sh-session
$ time npm run build

npm run build  7.43s user 5.82s system 221% cpu 5.985 total
```